### ISSUE TITLE
When building a Latex Lab Book with SCons, SCons does not build an index in a variant directory on the first build

### ISSUE THREAD

**Issue by bellockk** [SCons/scons#3878]
**Describe the bug**
https://stackoverflow.com/q/65986302/4704335

**Required information**
The issue was discussed in Discord.
* Version of SCons: v4.0.1
* Version of Python: 3.9.1
* Which python distribution if applicable (python.org, cygwin, anaconda, macports, brew,etc): Fedora 33 Package
* How you installed SCons: Fedora 33 Package Manager
* What Platform are you on? (Linux/Windows and which version): Fedora 33
* How to reproduce your issue?  Please include a small self contained reproducer. Likely a SConstruct should do for most issues.: https://github.com/bellockk/scons_issue_variantindex.git
* How you invoke scons (The command line you're using "scons --flags some_arguments"): "scons"

**Comment by bdbaddog** [comment]
@managan - could you take a look at this one?

**Comment by managan** [comment]
HI Bill,

I will try to take a look. Busy with a house sale however.

My first thought is I was surprised by the SConscript that looks like it does two builds that are “identical” except for the variant dir

SConscript(‘src/SConscript’, variant_dir=’build’)
SConscript(‘src/SConscript’)

Let me see if I can run it and reproduce his complaint.

Rob

*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-*-
Rob Managan
WCI/DP Division  LLNL
P.O. Box 808, L-095
Livermore, CA  94551-0808
Email: managan1@llnl.gov
Phone: 925-423-0903         FAX:   925-422-3389

On 1/31/21, 7:56 PM, "William Deegan" <notifications@github.com<mailto:notifications@github.com>> wrote:


@managan<https://github.com/managan> - could you take a look at this one?

**Comment by bdbaddog** [comment]
@managan - pretty sure @bellockk comments one then the other out to see the difference in how they run. He'd like it to work with variant dirs.

**Comment by bellockk** [comment]
Yes, the two SConscript lines are in the SConstruct to demonstrate how the `makeindex` call occurs in the build without the variant directory, but does not in the build with the variant directory.  What I would expect to see is the contents of the `src` folder and the `build` folder to be the same, but that is not the case.  It may be a clue that if you build the project as is, and then clean "scons && scons -c", scons will not clean up the `src/document.ilg` and `src/document.ind` files.  Instead of commenting out one of the SConscript() calls, when I want to build one way or the other I just execute scons with `scons src` or `scons build` to build just the version with the variant directory in the `build` folder or the version without the variant directory in the `src` folder.

**Comment by bdbaddog** [comment]
@bellockk - using the specified target on the command line `src` or `build` doesn't remove knowledge of said build targets from SCons. You should really comment out the one you're not trying to build in this situation.  

You can see this if you run `scons --tree=all src`  I believe.

**Comment by bellockk** [comment]
Understood, but I tried both.  The build steps remain the same in either case.
